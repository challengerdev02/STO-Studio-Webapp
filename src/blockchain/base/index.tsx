import { NearWalletConnectorState } from '../near/connector';
import {
  EVMWalletConnectorExtendedValues,
  EVMWalletConnectorState,
} from '../evm/connector';
import { useRouter } from 'next/router';
import { EVMProvider } from '../evm/provider';
import {
  Context,
  createContext,
  ReactNode,
  useEffect,
  useReducer,
} from 'react';
import { NearProtocolProvider } from '../near/provider';
import { isPlainObject } from 'lodash';
import { useDispatch } from 'react-redux';
import { Modal, Typography } from 'antd';
import { SUIWalletState, useSuiWallet } from '../sui';
import { SolanaWalletState, useSolanaWallet } from '../solana';
import { WalletContextState as SUIWalletContextState } from '@suiet/wallet-kit';
import { getBtcSeedSignature } from '../evm/utils';
import { sha256 } from '@/shared/utils';

export interface BaseProviderState {
  isConnected?: boolean;
  isConnecting?: boolean;
  showSignaturePrompt?: boolean;
  env?: string | null;
  near?: NearWalletConnectorState | null;
  evm?: EVMWalletConnectorState | null;
  sui?: ({ signedAddress?: string } & Partial<SUIWalletContextState>) | null;
  solana?: Partial<SolanaWalletState> | null;
}

export enum BaseProviderActionTypes {
  CONNECTOR = 'CONNECTOR',
  INITIATE = 'INITIATE',
  ERROR = 'ERROR',
  CONNECTED = 'CONNECTED',
  CONNECTING = 'CONNECTING',
  DISCONNECTED = 'DISCONNECTED',
  ACCOUNT_PROPERTIES = 'ACCOUNT_PROPERTIES',
  TOGGLE_PROMPT = 'TOGGLE_PROMPT',
}

export interface BaseProviderActions {
  type: BaseProviderActionTypes;
  payload: BaseProviderState;
}

const defaultState: BaseProviderState = {
  isConnected: false,
  isConnecting: false,
  env: null,
  near: null,
  evm: null,
  sui: null,
  solana: null,
};

export type MergeTypes<A, B, C> = {
  [key in keyof A]: key extends keyof B
    ? B[key]
    : key extends keyof C
    ? C[key]
    : A[key];
} & B &
  C;

export interface BaseProviderExtendedValues
  extends EVMWalletConnectorExtendedValues {}

type NearSUIStateMerge = MergeTypes<
  Partial<NearWalletConnectorState>,
  Partial<SUIWalletState>,
  Partial<SolanaWalletState>
>;

export interface BaseProviderContextValues
  extends Partial<EVMWalletConnectorState>,
    Pick<BaseProviderState, 'env' | 'isConnected' | 'isConnecting'>,
    BaseProviderExtendedValues,
    NearSUIStateMerge {
  connect: (...args: string[]) => Promise<void>;
  sign: (...args: any[]) => Promise<void>;
  disconnect: () => Promise<void>;
  getBalance: () => Promise<string>;
  unlockOrdinalWallet: (
    userInf: { walletAddress: string; user: string },
    ...args: any[]
  ) => Promise<any>;
}

export const BaseWeb3Context: Context<BaseProviderContextValues> =
  createContext<BaseProviderContextValues>({
    connect: async () => {},
    sign: async () => {},
    disconnect: async () => {},
    getBalance: async () => '0',
    unlockOrdinalWallet: async () => null,
    ...defaultState,
  });

interface BaseProviderProps {
  children: ReactNode | ReactNode[];
}

export type ReduxDispatch = ReturnType<typeof useDispatch>;

export const BaseProvider = (props: BaseProviderProps) => {
  const router = useRouter();
  const [state, dispatch] = useReducer(reducer, defaultState);
  const reduxDispatcher = useDispatch();

  const suiProvider = useSuiWallet(dispatch, reduxDispatcher);
  const solanaProvider = useSolanaWallet(dispatch, reduxDispatcher);
  const evmProvider = new EVMProvider(router, state, dispatch, reduxDispatcher);
  const nearProvider = new NearProtocolProvider(
    router,
    state,
    dispatch,
    reduxDispatcher
  );

  const connect = async (...args: any[]) => {
    switch (args[0]) {
      case 'near':
        await nearProvider.connect();
        break;

      case 'sui':
        suiProvider.connect(args[1]);
        break;

      case 'solana':
        solanaProvider.select(args[1]);
        break;

      case 'evm':
      default:
        await evmProvider.connect(args[1]);
        break;
    }
  };

  const disconnect = async () => {
    switch (state.env) {
      case 'near':
        await nearProvider.disconnect();
        break;

      case 'sui':
        await suiProvider.disconnect();
        break;

      case 'sui':
        await suiProvider.disconnect();
        break;

      case 'evm':
      default:
        await evmProvider.disconnect();
        break;
    }
  };

  const sign = async (...args: any[]) => {
    switch (state.env) {
      case 'near':
        await nearProvider.sign();
        break;

      case 'sui':
        await suiProvider.sign();
        break;

      case 'solana':
        await solanaProvider.sign({
          messageObject: args[1],
          message: args[2],
          result: args[3],
        });
        break;

      case 'evm':
      default:
        await evmProvider.sign(args[0]);
        break;
    }
  };

  const signMessage = async (...args: any[]) => {
    switch (state.env) {
      case 'near':
      //return await nearProvider.signMessage();

      case 'sui':
      //return await suiProvider.signMessage();

      case 'evm':
      default:
        return evmProvider.signMessage(args[0]);
    }
  };

  const unlockOrdinalWallet = async (
    info: { walletAddress: string; userId: string },
    ..._: any[]
  ) => {
    switch (state.env) {
      default:
        const message = getBtcSeedSignature({
          walletAddress: info.walletAddress,
          user: info.userId,
          environment: state.env,
        });
        return signMessage(message)
          .then((signature) => {
            return Buffer.from(sha256(signature), 'hex');
          })
          .catch((e) => {
            console.error(e);
            throw e;
          });
    }
  };

  const getBalance = async (): Promise<string> => {
    switch (state.env) {
      case 'near':
        return await nearProvider.getBalance();

      case 'evm':
      default:
        return await evmProvider.getBalance();
    }
  };

  const onToggleSignaturePrompt = (visibility: boolean = false) => {
    dispatch({
      type: BaseProviderActionTypes.TOGGLE_PROMPT,
      payload: { showSignaturePrompt: visibility },
    });
  };

  const onInit = async (): Promise<void> => {
    await nearProvider.onInit();
    await evmProvider.onInit();
  };

  useEffect(() => {
    nearProvider.detectConnection();
  }, [router.query]);

  useEffect(() => {
    onInit();
  }, []);

  useEffect(() => {
    evmProvider.removeListeners(state.evm?.provider);
    evmProvider.initializeListeners(state.evm?.provider);

    return () => {
      evmProvider.removeListeners(state.evm?.provider);
    };
  }, [state.evm?.provider]);

  // Add extension function [functions used by every provider]
  const extensions = { verifyEVMChain: evmProvider.verifyChain };
  // Create a joint state
  const _envState = Object.assign(
    {},
    state.near,
    state.evm,
    state.solana,
    suiProvider ? suiProvider : null
  );

  // console.log("nadafaf", s)
  const _values: BaseProviderContextValues = {
    ..._envState,
    connect,
    disconnect,
    sign,
    getBalance,
    ...state,
    signMessage,
    unlockOrdinalWallet,
    ...extensions,
  };

  return (
    <BaseWeb3Context.Provider value={_values}>
      <Modal
        visible={state?.showSignaturePrompt}
        onCancel={async () => {
          await _values.disconnect();
          onToggleSignaturePrompt(false);
        }}
        okText={'Accept and Sign'}
        cancelText={'Cancel'}
        okButtonProps={{
          shape: 'round',
        }}
        cancelButtonProps={{
          shape: 'round',
        }}
        onOk={() => _values.sign(_values?.chainChangeCallback)}
      >
        <Typography.Title>Signature Required!</Typography.Title>

        <span>
          By connecting your wallet and using {process.env.NEXT_PUBLIC_APP_NAME}
          , you agree to our{' '}
          <a
            href={
              process.env.NEXT_PUBLIC_TERMS_OF_SERVICE ??
              'https://metacomic.tawk.help/article/terms-of-service'
            }
            target={'_blank'}
            rel={'noreferrer'}
          >
            Terms of Service
          </a>{' '}
          and{' '}
          <a
            target={'_blank'}
            href={
              process.env.NEXT_PUBLIC_PRIVACY_POLICY ??
              'https://metacomic.tawk.help/article/privacy-policy'
            }
            rel={'noreferrer'}
          >
            Privacy Policy
          </a>
          .
        </span>
      </Modal>
      {props.children}
    </BaseWeb3Context.Provider>
  );
};

const appendReducerState = (
  state: Record<string, any>,
  incoming: Record<string, any>
): Record<string, any> => {
  const _state = { ...state };

  for (const prop in incoming) {
    if (isPlainObject(state[prop]) && isPlainObject(incoming[prop])) {
      _state[prop] = Object.assign({}, state[prop], incoming[prop]);
    } else {
      _state[prop] = incoming[prop];
    }
  }

  return _state;
};

export function reducer(state: BaseProviderState, action: BaseProviderActions) {
  const { type, payload } = action;
  switch (type) {
    case BaseProviderActionTypes.CONNECTOR:
    case BaseProviderActionTypes.INITIATE:
    case BaseProviderActionTypes.ERROR:
    case BaseProviderActionTypes.CONNECTED:
    case BaseProviderActionTypes.CONNECTING:
    case BaseProviderActionTypes.DISCONNECTED:
    case BaseProviderActionTypes.ACCOUNT_PROPERTIES:
    case BaseProviderActionTypes.TOGGLE_PROMPT:
      return appendReducerState(state, payload);
    default:
      return state;
  }
}
