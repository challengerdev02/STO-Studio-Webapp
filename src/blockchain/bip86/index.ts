import { networks } from 'bitcoinjs-lib';
import * as bip39 from 'bip39';
const bitcoinNetworks = {
  mainnet: networks.bitcoin,
  testnet: networks.testnet,
};
import { BIP32Factory } from 'bip32';
import ecurve from 'ecurve';
const secp256k1 = ecurve.getCurveByName('secp256k1');
import schnorr from 'bip-schnorr';
import { bech32 } from 'bech32';
import { bech32m } from 'bech32';
import { ecc } from '@/shared/utils/secp';
//import * as ecc from 'tiny-secp256k1';
// const ecc: any = {isPoint: ()=>false};
const bip32 = BIP32Factory(ecc);

export default class Bip86 {
  seed: any = undefined;
  isTestnet = false;
  coinType = 0;
  network: any = undefined;
  xprv: any = null;
  networks: any = bitcoinNetworks;
  xpub: any = null;
  /**
   * Constructor
   * Derive accounts from a mnemonic.
   * @param {string} mnemonic
   * @param {string} password
   * @param {boolean} isTestnet
   * @param {number} coinType - slip44
   * @param {object} network
   * @return
   */
  fromMnemonic(
    mnemonic: string,
    password = '',
    isTestnet?: boolean,
    network?: any,
    coinType?: any
  ): any {
    this.seed = bip39.mnemonicToSeedSync(mnemonic, password);
    this.isTestnet = isTestnet === true;
    this.coinType = this.isTestnet ? 1 : coinType ? coinType : 0; // 0 is for Bitcoin and 1 is testnet for all coins
    this.network =
      network || this.isTestnet
        ? bitcoinNetworks.testnet
        : bitcoinNetworks.mainnet;
    return this;
  }

  /**
   * Constructor
   * Derive accounts from seed.
   * @param {string} mnemonic
   * @param {string} password
   * @param {boolean} isTestnet
   * @param {number} coinType - slip44
   * @param {object} network
   * @return
   */
  fromSeed(
    seed: Buffer,
    isTestnet?: boolean,
    network?: any,
    coinType?: any
  ): any {
    this.seed = seed;
    this.isTestnet = isTestnet === true;
    this.coinType = this.isTestnet ? 1 : coinType ? coinType : 0; // 0 is for Bitcoin and 1 is testnet for all coins
    this.network =
      network || this.isTestnet
        ? bitcoinNetworks.testnet
        : bitcoinNetworks.mainnet;
    return this;
  }

  /**
   * Get root master private key
   * @return {string}
   */
  getRootPrivateKey(): string {
    let xprv = bip32.fromSeed(this.seed, this.network).toBase58();

    return xprv;
  }

  /**
   * Get root master public key
   * @return {string}
   */
  getRootPublicKey(): string {
    let xpub = bip32.fromSeed(this.seed, this.network).neutered().toBase58();

    return xpub;
  }

  /**
   * Derive a new master private key
   * @param {number} number
   * @param {number} changePurpose
   * @return {string}
   */
  deriveAccount(number: number, changePurpose: number): string {
    let purpose = changePurpose || 86,
      keypath = 'm/' + purpose + "'/" + this.coinType + "'/" + number + "'",
      account = bip32
        .fromSeed(this.seed, this.network)
        .derivePath(keypath)
        .toBase58();

    return account;
  }

  /**
   * Constructor
   * Create key pairs from a private master key of mainnet and testnet.
   * @param {string} xprv/tprv
   * @param {object} networks
   */
  fromXPrv(xprv: string, networks?: any) {
    this.networks = networks || bitcoinNetworks;
    this.xprv = xprv;
    this.getNetworkFromPrivKey(xprv);
    return this;
  }

  getNetworkFromPrivKey(xprv: any) {
    let key = xprv.slice(0, 4);

    if (key !== 'xprv' && key !== 'tprv') {
      throw new Error('prefix is not supported');
    }

    if (key === 'xprv') {
      this.network = this.networks.mainnet;
      this.isTestnet = false;
    }

    if (key === 'tprv') {
      this.network = this.networks.testnet;
      this.isTestnet = true;
    }
  }

  /**
   * Get account master private key
   * @return {string}
   */
  getAccountPrivateKey() {
    let xprv = bip32.fromBase58(this.xprv, this.network).toBase58();

    return xprv;
  }

  /**
   * Get account master public key
   * @return {string}
   */
  getAccountPublicKeyFromPrivKey() {
    let xpub = bip32.fromBase58(this.xprv, this.network).neutered().toBase58();

    return xpub;
  }

  /**
   * Get private key
   * @param {number} index
   * @param {boolean} isChange
   * @return {string}
   */
  getPrivateKey(index: number, isChange: boolean) {
    let change = isChange === true ? 1 : 0,
      prvkey = bip32
        .fromBase58(this.xprv, this.network)
        .derive(change)
        .derive(index);

    return prvkey.toWIF();
  }

  /**
   * Get public key
   * @param {number} index
   * @param {boolean} isChange
   * @return {string}
   */
  getPublicKeyFromPrivKey(index: number, isChange: boolean) {
    let change = isChange === true ? 1 : 0,
      pubKey = bip32
        .fromBase58(this.xprv, this.network)
        .derive(change)
        .derive(index).publicKey;

    return pubKey.toString('hex');
  }

  /**
   * Get address
   * @param {number} index
   * @param {boolean} isChange
   * @param {number} purpose
   * @return {string}
   */

  getAddressFromPrivateKey(index: number, isChange: boolean, _?: number) {
    let change = isChange === true ? 1 : 0,
      pubkey = bip32
        .fromBase58(this.xprv, this.network)
        .derive(change)
        .derive(index).publicKey;

    return this.getP2TRAddress(pubkey, this.isTestnet);
  }

  /**
   * Constructor
   * Create public keys and addresses from a public master key of mainnet and testnet.
   * @param {string} xpub/tpub
   * @param {object} networks
   */
  fromXPub(xpub: string, networks: any) {
    this.networks = networks || bitcoinNetworks;
    this.xpub = xpub;
    this.getNetworkFromPubKey(xpub);
  }

  getNetworkFromPubKey(xpub: any) {
    let key = xpub.slice(0, 4);

    if (key !== 'xpub' && key !== 'tpub') {
      throw new Error('prefix is not supported');
    }

    if (key === 'xpub') {
      this.network = this.networks.mainnet;
      this.isTestnet = false;
    }

    if (key === 'tpub') {
      this.network = this.networks.testnet;
      this.isTestnet = true;
    }
  }

  /**
   * Get account master public key
   * @return {string}
   */
  getAccountPublicKeyFromPubKey() {
    let xpub = bip32.fromBase58(this.xpub, this.network).neutered().toBase58();

    return xpub;
  }

  /**
   * Get public key
   * @param {number} index
   * @param {boolean} isChange
   * @return {string}
   */
  getPublicKeyFromPubKey(index: number, isChange: boolean) {
    let change = isChange === true ? 1 : 0,
      xpub = bip32
        .fromBase58(this.xpub, this.network)
        .derive(change)
        .derive(index);

    return xpub.publicKey.toString('hex');
  }

  /**
   * Get address
   * @param {number} index
   * @param {boolean} isChange
   * @param {number} purpose
   * @return {string}
   */
  public getAddressFromPubKey(index: number, isChange: boolean, _?: string) {
    let change = isChange === true ? 1 : 0,
      pubkey = bip32
        .fromBase58(this.xpub, this.network)
        .derive(change)
        .derive(index).publicKey;

    return this.getP2TRAddress(pubkey, this.isTestnet);
  }

  public getP2TRAddress(pubkey: Buffer, testnet = false) {
    const pubKey = ecurve.Point.decodeFrom(secp256k1, pubkey);
    const taprootPubkey = schnorr.taproot.taprootConstruct(pubKey);
    const words = bech32.toWords(taprootPubkey);
    words.unshift(1);
    return bech32m.encode(testnet ? 'tb' : 'bc', words);
  }
}
