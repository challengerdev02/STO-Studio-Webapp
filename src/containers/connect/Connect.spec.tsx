import { BaseWeb3Context } from '../../blockchain/base';
import { cleanup, render } from '@testing-library/react';
import { ConnectWalletContainer } from './index';

const mockPush = jest.fn();
const useRouter = jest.spyOn(require('next/router'), 'useRouter');

// jest.mock('next/router', () => ({
//     ...jest.requireActual('next/router'), // use actual for all non-hook parts
//     useParams: () => ({
//       teamId: 'someTeamId',
//       eventId: 'someEventId',
//       broadcastId: 'someShowId',
//     }),
//     useRouter: () => ({
//       query: {},
//       push: mockPush,
//     }),
//     useLocation: jest.fn(),
//   }));
//   import { useRouter } from 'next/router';

describe('View Account Connect', function () {
  afterEach(cleanup);
  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  const ProviderEl = ({ iscon = true }) => {
    return (
      <BaseWeb3Context.Provider
        value={{
          isConnected: iscon,
          isConnecting: true,
          provider: true,
        }}
      >
        <ConnectWalletContainer />
      </BaseWeb3Context.Provider>
    );
  };
  it('should match with snapshot', () => {
    // @ts-ignore

    useRouter.mockImplementation(() => ({
      query: {},
      push: mockPush,
    }));
    const { container } = render(<ProviderEl />);
    expect(container).toMatchSnapshot();
  });

  it('should match with snapshot', () => {
    // @ts-ignore

    useRouter.mockImplementation(() => ({
      query: {},
      push: mockPush,
    }));
    const { container } = render(<ProviderEl iscon={false} />);
    expect(container).toMatchSnapshot();
  });

  it('should match with snapshot With referrer ', () => {
    // @ts-ignore
    useRouter.mockImplementation(() => ({
      query: { referrer: '/s' },
      push: mockPush,
    }));

    const { container } = render(<ProviderEl />);
    expect(container).toMatchSnapshot();
  });
});
