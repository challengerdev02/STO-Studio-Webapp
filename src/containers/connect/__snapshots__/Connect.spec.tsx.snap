// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Account Connect should match with snapshot 1`] = `
<div>
  <div
    class="sc-jYmNlR icaNaC meta-flex meta-flex-col meta-flex-j-c meta-align-center"
  >
    <h2
      class="ant-typography"
    >
      Connect your wallet
    </h2>
    <h5
      class="ant-typography"
    >
      Choose one of available wallet providers or create a new wallet.
       
      <a
        class="ant-typography"
      >
        What is a wallet?
      </a>
    </h5>
    <br />
    <br />
    <div
      class="ant-space ant-space-vertical connect-wallet-buttons-container"
    >
      <div
        class="ant-space-item"
        style="margin-bottom: 20px;"
      >
        <div
          style="opacity: 0; transform: scale(0.95) translateZ(0);"
        >
          <button
            class="ant-btn ant-btn-round ant-btn-primary ant-btn-block"
            disabled=""
            type="button"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center w-100 meta-flex-s-b meta-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <img
                  alt="Metamask"
                  height="24"
                  src="/test-file-stub"
                  width="24"
                />
              </div>
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <span>
                  Sign in with 
                  Metamask
                </span>
              </div>
              <div
                class="ant-space-item"
              >
                <span />
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        class="ant-space-item"
        style="margin-bottom: 20px;"
      >
        <div
          style="opacity: 0; transform: scale(0.95) translateZ(0);"
        >
          <button
            class="ant-btn ant-btn-round ant-btn-default ant-btn-block"
            disabled=""
            type="button"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center w-100 meta-flex-s-b meta-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <img
                  alt="Coinbase"
                  height="24"
                  src="/test-file-stub"
                  width="24"
                />
              </div>
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <span>
                  
                  Coinbase
                </span>
              </div>
              <div
                class="ant-space-item"
              >
                <span />
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        class="ant-space-item"
        style="margin-bottom: 20px;"
      >
        <div
          style="opacity: 0; transform: scale(0.95) translateZ(0);"
        >
          <button
            class="ant-btn ant-btn-round ant-btn-default ant-btn-block"
            disabled=""
            type="button"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center w-100 meta-flex-s-b meta-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <img
                  alt="Mobile Wallet"
                  height="24"
                  src="/test-file-stub"
                  width="24"
                />
              </div>
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <span>
                  
                  Mobile Wallet
                </span>
              </div>
              <div
                class="ant-space-item"
              >
                <span />
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        class="ant-space-item"
        style="margin-bottom: 20px;"
      >
        <div
          style="opacity: 0; transform: scale(0.95) translateZ(0);"
        >
          <button
            class="ant-btn ant-btn-round ant-btn-default ant-btn-block"
            disabled=""
            type="button"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center w-100 meta-flex-s-b meta-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <img
                  alt="Binance Chain"
                  height="24"
                  src="/test-file-stub"
                  width="24"
                />
              </div>
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <span>
                  
                  Binance Chain
                </span>
              </div>
              <div
                class="ant-space-item"
              >
                <span />
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        class="ant-space-item"
      >
        <div
          style="opacity: 0;"
        >
          <button
            class="ant-btn ant-btn-round ant-btn-default ant-btn-block"
            style="justify-content: center;"
            type="button"
          >
            <span>
              Show more options
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`View Account Connect should match with snapshot 2`] = `
<div>
  <div
    class="sc-jYmNlR icaNaC meta-flex meta-flex-col meta-flex-j-c meta-align-center"
  >
    <h2
      class="ant-typography"
    >
      Connect your wallet
    </h2>
    <h5
      class="ant-typography"
    >
      Choose one of available wallet providers or create a new wallet.
       
      <a
        class="ant-typography"
      >
        What is a wallet?
      </a>
    </h5>
    <br />
    <br />
    <div
      class="ant-space ant-space-vertical connect-wallet-buttons-container"
    >
      <div
        class="ant-space-item"
        style="margin-bottom: 20px;"
      >
        <div
          data-projection-id="1"
          style="opacity: 0; transform: scale(0.95) translateZ(0);"
        >
          <button
            class="ant-btn ant-btn-round ant-btn-primary ant-btn-block"
            disabled=""
            type="button"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center w-100 meta-flex-s-b meta-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <img
                  alt="Metamask"
                  height="24"
                  src="/test-file-stub"
                  width="24"
                />
              </div>
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <span>
                  Sign in with 
                  Metamask
                </span>
              </div>
              <div
                class="ant-space-item"
              >
                <span />
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        class="ant-space-item"
        style="margin-bottom: 20px;"
      >
        <div
          data-projection-id="2"
          style="opacity: 0; transform: scale(0.95) translateZ(0);"
        >
          <button
            class="ant-btn ant-btn-round ant-btn-default ant-btn-block"
            disabled=""
            type="button"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center w-100 meta-flex-s-b meta-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <img
                  alt="Coinbase"
                  height="24"
                  src="/test-file-stub"
                  width="24"
                />
              </div>
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <span>
                  
                  Coinbase
                </span>
              </div>
              <div
                class="ant-space-item"
              >
                <span />
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        class="ant-space-item"
        style="margin-bottom: 20px;"
      >
        <div
          data-projection-id="3"
          style="opacity: 0; transform: scale(0.95) translateZ(0);"
        >
          <button
            class="ant-btn ant-btn-round ant-btn-default ant-btn-block"
            disabled=""
            type="button"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center w-100 meta-flex-s-b meta-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <img
                  alt="Mobile Wallet"
                  height="24"
                  src="/test-file-stub"
                  width="24"
                />
              </div>
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <span>
                  
                  Mobile Wallet
                </span>
              </div>
              <div
                class="ant-space-item"
              >
                <span />
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        class="ant-space-item"
        style="margin-bottom: 20px;"
      >
        <div
          data-projection-id="4"
          style="opacity: 0; transform: scale(0.95) translateZ(0);"
        >
          <button
            class="ant-btn ant-btn-round ant-btn-default ant-btn-block"
            disabled=""
            type="button"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center w-100 meta-flex-s-b meta-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <img
                  alt="Binance Chain"
                  height="24"
                  src="/test-file-stub"
                  width="24"
                />
              </div>
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <span>
                  
                  Binance Chain
                </span>
              </div>
              <div
                class="ant-space-item"
              >
                <span />
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        class="ant-space-item"
      >
        <div
          data-projection-id="5"
          style="opacity: 0;"
        >
          <button
            class="ant-btn ant-btn-round ant-btn-default ant-btn-block"
            style="justify-content: center;"
            type="button"
          >
            <span>
              Show more options
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`View Account Connect should match with snapshot With referrer  1`] = `
<div>
  <div
    class="sc-jYmNlR icaNaC meta-flex meta-flex-col meta-flex-j-c meta-align-center"
  >
    <h2
      class="ant-typography"
    >
      Connect your wallet
    </h2>
    <h5
      class="ant-typography"
    >
      Choose one of available wallet providers or create a new wallet.
       
      <a
        class="ant-typography"
      >
        What is a wallet?
      </a>
    </h5>
    <br />
    <br />
    <div
      class="ant-space ant-space-vertical connect-wallet-buttons-container"
    >
      <div
        class="ant-space-item"
        style="margin-bottom: 20px;"
      >
        <div
          data-projection-id="6"
          style="opacity: 0; transform: scale(0.95) translateZ(0);"
        >
          <button
            class="ant-btn ant-btn-round ant-btn-primary ant-btn-block"
            disabled=""
            type="button"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center w-100 meta-flex-s-b meta-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <img
                  alt="Metamask"
                  height="24"
                  src="/test-file-stub"
                  width="24"
                />
              </div>
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <span>
                  Sign in with 
                  Metamask
                </span>
              </div>
              <div
                class="ant-space-item"
              >
                <span />
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        class="ant-space-item"
        style="margin-bottom: 20px;"
      >
        <div
          data-projection-id="7"
          style="opacity: 0; transform: scale(0.95) translateZ(0);"
        >
          <button
            class="ant-btn ant-btn-round ant-btn-default ant-btn-block"
            disabled=""
            type="button"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center w-100 meta-flex-s-b meta-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <img
                  alt="Coinbase"
                  height="24"
                  src="/test-file-stub"
                  width="24"
                />
              </div>
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <span>
                  
                  Coinbase
                </span>
              </div>
              <div
                class="ant-space-item"
              >
                <span />
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        class="ant-space-item"
        style="margin-bottom: 20px;"
      >
        <div
          data-projection-id="8"
          style="opacity: 0; transform: scale(0.95) translateZ(0);"
        >
          <button
            class="ant-btn ant-btn-round ant-btn-default ant-btn-block"
            disabled=""
            type="button"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center w-100 meta-flex-s-b meta-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <img
                  alt="Mobile Wallet"
                  height="24"
                  src="/test-file-stub"
                  width="24"
                />
              </div>
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <span>
                  
                  Mobile Wallet
                </span>
              </div>
              <div
                class="ant-space-item"
              >
                <span />
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        class="ant-space-item"
        style="margin-bottom: 20px;"
      >
        <div
          data-projection-id="9"
          style="opacity: 0; transform: scale(0.95) translateZ(0);"
        >
          <button
            class="ant-btn ant-btn-round ant-btn-default ant-btn-block"
            disabled=""
            type="button"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center w-100 meta-flex-s-b meta-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <img
                  alt="Binance Chain"
                  height="24"
                  src="/test-file-stub"
                  width="24"
                />
              </div>
              <div
                class="ant-space-item"
                style="margin-right: 10px;"
              >
                <span>
                  
                  Binance Chain
                </span>
              </div>
              <div
                class="ant-space-item"
              >
                <span />
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        class="ant-space-item"
      >
        <div
          data-projection-id="10"
          style="opacity: 0;"
        >
          <button
            class="ant-btn ant-btn-round ant-btn-default ant-btn-block"
            style="justify-content: center;"
            type="button"
          >
            <span>
              Show more options
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
